packages/server/src/handlers/chat.ts

```diff
@@ -1,54 +1,66 @@
-import type { AgentInputData } from "@llamaindex/workflow";
+import { stopAgentEvent } from "@llamaindex/workflow";
 import { type Message } from "ai";
 import { IncomingMessage, ServerResponse } from "http";
 import type { MessageType } from "llamaindex";
 import { type WorkflowFactory } from "../types";
+import { sendSuggestedQuestionsEvent } from "../utils";
+import {
+  getHumanResponsesFromMessage,
+  pauseForHumanInput,
+} from "../utils/hitl";
 import {
   parseRequestBody,
   pipeStreamToResponse,
   sendJSONResponse,
 } from "../utils/request";
 import { toDataStream } from "../utils/stream";
-import { sendSuggestedQuestionsEvent } from "../utils/suggestion";
-import { runWorkflow } from "../utils/workflow";
+import { processWorkflowStream, runWorkflow } from "../utils/workflow";
 
 export const handleChat = async (
   req: IncomingMessage,
   res: ServerResponse,
   workflowFactory: WorkflowFactory,
   suggestNextQuestions: boolean,
 ) => {
+  const abortController = new AbortController();
+  res.on("close", () => abortController.abort("Connection closed"));
+
   try {
     const body = await parseRequestBody(req);
-    const { messages } = body as { messages: Message[] };
-    const chatHistory = messages.map((message) => ({
-      role: message.role as MessageType,
-      content: message.content,
-    }));
+    const { messages, id: requestId } = body as {
+      messages: Message[];
+      id?: string;
+    };
 
     const lastMessage = messages[messages.length - 1];
-    if (lastMessage?.role !== "user") {
+    if (lastMessage?.role !== "user" || !lastMessage.content) {
       return sendJSONResponse(res, 400, {
         error: "Messages cannot be empty and last message must be from user",
       });
     }
-    const workflowInput: AgentInputData = {
-      userInput: lastMessage.content,
-      chatHistory,
-    };
 
-    const abortController = new AbortController();
-    res.on("close", () => abortController.abort("Connection closed"));
+    const chatHistory = messages.map((message) => ({
+      role: message.role as MessageType,
+      content: message.content,
+    }));
+
+    const context = await runWorkflow({
+      workflow: await workflowFactory(body),
+      input: { userInput: lastMessage.content, chatHistory },
```

Comment:
`userInput` is `MessageContentDetail` but `lastMessage` is just a vercel message, so it's just a string now?